### PR TITLE
Add search page and interview page functionality in frontend

### PR DESCRIPTION
**Pull request by ongweisheng**
Search Page
- 30 second timer shown while finding match
- Users will be redirected back to home if no match is found within 30 seconds
- Users will be redirected back to home if they cancel find match. A window will pop up to ask them to confirm if they want to cancel finding match when clicking the cancel find match button. Upon confirmation in the window popup, users will be redirected back to home
- Users will be redirected to interview page if a successful match is found

Interview Page
- Question of type difficulty that was selected by the matched users will be shown on the screen
- Users will be able to end the interview by clicking the end interview button. A window popup will then be shown asking the users to confirm again whether they want to end the interview. Users will then be redirected back to home upon confirmation

### CODE REVIEW

**Comment by TheSpaceCuber** [comment]
Will take a look on Sunday 2nd Oct

**Comment by Herrekt** [comment]
When the interview is ended via postman, the page on the front end will freeze on the interview page and the end interview button does not work.
The find match button can be clicked without selecting difficulty, and the page cannot be exited via cancel find match request.
Had an error I could not replicate. Shown below:
Crashed

File: frontend/src/components/InterviewPage.js
Comment (TheSpaceCuber) [inline frontend/src/components/InterviewPage.js:33]: Suggest to have 1 line spacing

File: frontend/src/components/InterviewPage.js
Comment (TheSpaceCuber) [inline frontend/src/components/InterviewPage.js:61]: Suggest renaming this button to `Cancel` or `Resume` or `Go Back`. 

onClick should not be `handleEndFindMatch` but `handleCloseEndInterview` too I believe. Can consider renaming state variables or function names to something like showEndInterviewPrompt (or maybe another name). Others can chip in on naming as well.

File: frontend/src/services/matchingService.js
Comment (TheSpaceCuber) [inline frontend/src/services/matchingService.js:21]: Can consider 1 line return statement instead? 

Not sure if we need try-catch block https://stackoverflow.com/questions/7148019/when-should-you-use-try-catch-in-javascript

but maybe can add if statement checks instead just incase of a null response in that case? (means no more 1 line return statement)

**Review by TheSpaceCuber — CHANGES_REQUESTED** [review]
LGTM with minor suggestions. Decent UI so far. Might need to wait for the user service PR merge to fully test end to end before merging.

**Comment by ongweisheng** [comment]
> When the interview is ended via postman, the page on the front end will freeze on the interview page and the end interview button does not work. The find match button can be clicked without selecting difficulty, and the page cannot be exited via cancel find match request. Had an error I could not replicate. Shown below: During a sent match request from postman and frontend. Noticed it only when the frontend couldn't cancel the match request.
> 
> Crashed

This issue will most probably be fixed by Royce PR as discussed and I believe the changes have been made already as well

File: frontend/src/components/InterviewPage.js
Comment (ongweisheng) [reply frontend/src/components/InterviewPage.js:9]: Done, will be inserted back if required. Removed in the other pages as well.

File: frontend/src/components/InterviewPage.js
Comment (ongweisheng) [reply frontend/src/components/InterviewPage.js:33]: Reason for no 1 line spacing is because this 3 functions are grouped together and required for the modal. Grouping together would be clearer to show their purpose together

File: frontend/src/components/InterviewPage.js
Comment (ongweisheng) [reply frontend/src/components/InterviewPage.js:61]: Have renamed button to `Resume interview`. Also for the onClick as shown on top have changed it to `handleCloseEndInterview`. Renamed the function to end the interview as `handleEndInterview` as well.

File: frontend/src/services/matchingService.js
Comment (ongweisheng) [reply frontend/src/services/matchingService.js:21]: Should backend do the null checking or frontend do the null checking?

Comment (TheSpaceCuber) [reply frontend/src/services/matchingService.js:21]: Both would be good.

**Review by TheSpaceCuber — APPROVED** [review]
LGTM.

**Pull request merged**
